src/R/Editor/Impl/Settings/REditorSettings.cs

```diff
@@ -22,123 +23,111 @@ public static class REditorSettings {
         public const string PartialArgumentNameMatchKey = "PartialArgumentNameMatch";
         public const string EnableOutliningKey = "EnableOutlining";
 
-        private static bool _initialized = false;
+        private static IEditorSettingsStorage _storage;
         private static RFormatOptions _formatOptions = new RFormatOptions();
 
-        private static IEditorSettingsStorage Storage {
-            get {
-                var storage = EditorShell.GetSettings(EditorShell.Current, RContentTypeDefinition.LanguageName);
-
-                if (!_initialized) {
-                    storage.SettingsChanged += OnSettingsChanged;
-                    _initialized = true;
-                }
-
-                return storage;
-            }
-        }
-
-        public static IWritableEditorSettingsStorage WritableStorage {
-            get { return Storage as IWritableEditorSettingsStorage; }
+        public static void Initialize(ICoreShell coreShell) {
```

Comment:
Yes for now. I will turn editor settings into a separate object and attach to the document/buffer as a service when `ServiceManager` static will be eliminated